src/ott/neural/flow_models/otfm.py

```diff
@@ -0,0 +1,199 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple
+
+import jax
+import jax.numpy as jnp
+import jax.tree_util as jtu
+import numpy as np
+
+import diffrax
+from flax.training import train_state
+
+from ott import utils
+from ott.neural.flow_models import flows, models
+from ott.neural.flow_models import utils as flow_utils
+
+__all__ = ["OTFlowMatching"]
+
+
+class OTFlowMatching:
+  """(Optimal transport) flow matching :cite:`lipman:22`.
+
+  With an extension to OT-FM :cite:`tong:23`, :cite:`pooladian:23`.
+
+  Args:
+    velocity_field: Neural vector field parameterized by a neural network.
+    flow: Flow between source and target distribution.
+    match_fn: Function to match data points from the source distribution and
+      the target distribution.
+    time_sampler: Sampler for the time.
+    # TODO(michalk8): expose all args for the train state?
+    kwargs: TODO.
+  """
+
+  # TODO(michalk8): in the future, `input_dim`, `optimizer` and `rng` will be
```

Comment:
Similarly as in `GENOT`, let's expose it for now. In the upcoming PR, `__call__` will perform just 1 step + we will define trainers.